encoding/src/adapters/rle_lossless.rs

```diff
@@ -0,0 +1,198 @@
+use byteordered::byteorder::{ByteOrder, LittleEndian};
+use snafu::OptionExt;
+
+use crate::adapters::{DecodeResult, EncodeResult, PixelDataObject, PixelRWAdapter};
+use std::io::{self, Read, Seek};
+
+use super::{CustomDecodeError, MissingAttribute};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct RLELosslessAdapter {}
```

Comment:
Can be a unit struct.
```suggestion
pub struct RLELosslessAdapter;
```